lib/jwt/signature.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+require 'openssl'
+
+module JWT
+  module Signature
+    extend self
```

Comment:
what about `private` functions? I don't think it's possible to define private function with `module_functions`. 
What are the benefits of `module_functions`?